onnxruntime/core/providers/cpu/nn/batch_norm_helper.h

```diff
@@ -118,8 +118,8 @@ class BatchNormHelper {
     auto rank = x_shape.NumDimensions();
     auto num_samples = rank > 0 ? orig_dims[0] : 1;  // NCHW
     auto num_channels = rank > 1 ? orig_dims[1] : 1;
-    auto width = rank > 3 ? orig_dims[3] : 1;
     auto height = rank > 2 ? orig_dims[2] : 1;
+    int64_t width = 1;
```

Comment:
Does this means that we don't support spatial batchnorm?

Reply:
When orig_dims.size() is >= 4, we already returned above. When the code goes here, since orig_dims.size() <= 3, then for N*C*H*W, the W must be 1. Above "rank > 3 ? orig_dims[3] : 1;" is useless because rank > 3 is not possible. I am guessing this piece of code has nothing to do with spatial?

Reply:
And actually spatial is the default and currently the only mode in this implementation and also in the [CPU BN](https://github.com/microsoft/onnxruntime/blob/4275055868359e41f436c2b170f03d2a52fda057/onnxruntime/core/providers/cpu/nn/batch_norm.h#L52). A new attribute `spatial` might be needed to specify the mode.

Reply:
OK. thanks for the explaination!